### ISSUE TITLE
MusicXML: notes after clef change not displayed correctly

### ISSUE THREAD

**Issue by phboer** [rism-ch/verovio#540]
Expected:
![image](https://cloud.githubusercontent.com/assets/25896800/24929161/52caf594-1f05-11e7-97ae-a3f37164ea52.png)

Shown by Verovio:
![image](https://cloud.githubusercontent.com/assets/25896800/24929184/6ca17d8a-1f05-11e7-9a48-f208e05d7dcd.png)

The notes after the clef change are not displayed correctly. 

MusicXML fragement of the measure that contains the clef change:
```
    <measure number="18" width="160.91">
      <attributes>
        <clef number="2">
          <sign>F</sign>
          <line>4</line>
          </clef>
        </attributes>
      <note default-x="12.00" default-y="-25.00">
        <pitch>
          <step>A</step>
          <octave>4</octave>
          </pitch>
        <duration>12</duration>
        <voice>1</voice>
        <type>eighth</type>
        <stem>up</stem>
        <staff>1</staff>
        </note>
      <note>
        <rest/>
        <duration>6</duration>
        <voice>1</voice>
        <type>16th</type>
        <staff>1</staff>
        </note>
      <note default-x="85.65" default-y="-50.00">
        <pitch>
          <step>C</step>
          <octave>4</octave>
          </pitch>
        <duration>6</duration>
        <voice>1</voice>
        <type>16th</type>
        <stem>up</stem>
        <staff>1</staff>
        <beam number="1">begin</beam>
        <beam number="2">begin</beam>
        </note>
      <note default-x="110.20" default-y="-40.00">
        <pitch>
          <step>E</step>
          <octave>4</octave>
          </pitch>
        <duration>6</duration>
        <voice>1</voice>
        <type>16th</type>
        <stem>up</stem>
        <staff>1</staff>
        <beam number="1">continue</beam>
        <beam number="2">continue</beam>
        </note>
      <note default-x="134.76" default-y="-25.00">
        <pitch>
          <step>A</step>
          <octave>4</octave>
          </pitch>
        <duration>6</duration>
        <voice>1</voice>
        <type>16th</type>
        <stem>up</stem>
        <staff>1</staff>
        <beam number="1">end</beam>
        <beam number="2">end</beam>
        </note>
      <backup>
        <duration>36</duration>
        </backup>
      <direction placement="below">
        <direction-type>
          <pedal type="start" line="yes"/>
          </direction-type>
        <staff>2</staff>
        </direction>
      <note default-x="12.00" default-y="-140.00">
        <pitch>
          <step>A</step>
          <octave>2</octave>
          </pitch>
        <duration>6</duration>
        <voice>5</voice>
        <type>16th</type>
        <stem>down</stem>
        <staff>2</staff>
        <beam number="1">begin</beam>
        <beam number="2">begin</beam>
        </note>
      <note default-x="36.55" default-y="-120.00">
        <pitch>
          <step>E</step>
          <octave>3</octave>
          </pitch>
        <duration>6</duration>
        <voice>5</voice>
        <type>16th</type>
        <stem>down</stem>
        <staff>2</staff>
        <beam number="1">continue</beam>
        <beam number="2">continue</beam>
        </note>
      <note default-x="61.10" default-y="-105.00">
        <pitch>
          <step>A</step>
          <octave>3</octave>
          </pitch>
        <duration>6</duration>
        <voice>5</voice>
        <type>16th</type>
        <stem>down</stem>
        <staff>2</staff>
        <beam number="1">end</beam>
        <beam number="2">end</beam>
        </note>
      <note>
        <rest/>
        <duration>6</duration>
        <voice>5</voice>
        <type>16th</type>
        <staff>2</staff>
        </note>
      <note>
        <rest/>
        <duration>12</duration>
        <voice>5</voice>
        <type>eighth</type>
        <staff>2</staff>
        </note>
      </measure>
```

**Comment by rettinghaus** [comment]
It would be helpful if you could provide the complete MusicXML file or the MEI ouput by Verovio.

**Comment by craigsapp** [comment]
Also you should report what version of verovio you are using such as which branch of the repository (master, develop, etc),  or also post the MEI file which will include that information.

**Comment by craigsapp** [comment]
Here is one way of expressing the music as two parts.

Rendered in Finale:

<img width="707" alt="screen shot 2017-04-14 at 12 16 07 pm" src="https://cloud.githubusercontent.com/assets/3487289/25053469/498488b6-210c-11e7-9b7a-2508a60220b0.png">

Rendered into notation in veorvio:

<img width="1070" alt="screen shot 2017-04-14 at 12 17 45 pm" src="https://cloud.githubusercontent.com/assets/3487289/25053481/64d539a8-210c-11e7-8c0f-dc8ec8caf7a3.png">

With the command:

```bash
verovio -f musicxml file.xml
```

The problem in this case is that the clef change is on the wrong side of the barline from how it is displayed in Finale.

MusicXML test data:

```xml
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<!DOCTYPE score-partwise PUBLIC "-//Recordare//DTD MusicXML 3.0 Partwise//EN" "http://www.musicxml.org/dtds/partwise.dtd">
<score-partwise version="3.0">
  <movement-title>Fury lease</movement-title>
  <identification>
    <creator type="composer">Louis Beetfield</creator>
    <encoding>
      <software>Finale v25 for Mac</software>
      <encoding-date>2017-04-14</encoding-date>
      <supports attribute="new-system" element="print" type="yes" value="yes"/>
      <supports attribute="new-page" element="print" type="yes" value="yes"/>
      <supports element="accidental" type="yes"/>
      <supports element="beam" type="yes"/>
      <supports element="stem" type="yes"/>
    </encoding>
  </identification>
  <defaults>
    <scaling>
      <millimeters>7.2319</millimeters>
      <tenths>40</tenths>
    </scaling>
    <page-layout>
      <page-height>1545</page-height>
      <page-width>1194</page-width>
      <page-margins type="both">
        <left-margin>70</left-margin>
        <right-margin>70</right-margin>
        <top-margin>88</top-margin>
        <bottom-margin>88</bottom-margin>
      </page-margins>
    </page-layout>
    <system-layout>
      <system-margins>
        <left-margin>0</left-margin>
        <right-margin>0</right-margin>
      </system-margins>
      <system-distance>121</system-distance>
      <top-system-distance>70</top-system-distance>
    </system-layout>
    <staff-layout>
      <staff-distance>70</staff-distance>
    </staff-layout>
    <appearance>
      <line-width type="stem">0.7487</line-width>
      <line-width type="beam">5</line-width>
      <line-width type="staff">0.7487</line-width>
      <line-width type="light barline">0.7487</line-width>
      <line-width type="heavy barline">5</line-width>
      <line-width type="leger">0.7487</line-width>
      <line-width type="ending">0.7487</line-width>
      <line-width type="wedge">0.7487</line-width>
      <line-width type="enclosure">0.7487</line-width>
      <line-width type="tuplet bracket">0.7487</line-width>
      <note-size type="grace">60</note-size>
      <note-size type="cue">60</note-size>
      <distance type="hyphen">120</distance>
      <distance type="beam">8</distance>
    </appearance>
    <music-font font-family="Maestro,engraved" font-size="20.5"/>
    <word-font font-family="Times New Roman" font-size="10.25"/>
  </defaults>
  <credit page="1">
    <credit-type>title</credit-type>
    <credit-words default-x="597" default-y="1440" font-size="24" justify="center" valign="top">Fury lease</credit-words>
  </credit>
  <credit page="1">
    <credit-type>composer</credit-type>
    <credit-words default-x="1124" default-y="1362" font-size="12" justify="right" valign="top">Louis Beetfield</credit-words>
  </credit>
  <credit page="1">
    <credit-words default-x="70" default-y="1453" font-size="12" valign="top">Score</credit-words>
  </credit>
  <part-list>
    <part-group number="1" type="start">
      <group-symbol default-x="-5">brace</group-symbol>
      <group-barline>yes</group-barline>
    </part-group>
    <score-part id="P1">
      <part-name print-object="no">MusicXML Part</part-name>
      <score-instrument id="P1-I1">
        <instrument-name>SmartMusic SoftSynth 1</instrument-name>
      </score-instrument>
      <midi-instrument id="P1-I1">
        <midi-channel>1</midi-channel>
        <midi-bank>15489</midi-bank>
        <midi-program>1</midi-program>
        <volume>80</volume>
        <pan>0</pan>
      </midi-instrument>
    </score-part>
    <score-part id="P2">
      <part-name print-object="no">MusicXML Part</part-name>
      <score-instrument id="P2-I2">
        <instrument-name>SmartMusic SoftSynth 1</instrument-name>
      </score-instrument>
      <midi-instrument id="P2-I2">
        <midi-channel>2</midi-channel>
        <midi-bank>15489</midi-bank>
        <midi-program>1</midi-program>
        <volume>80</volume>
        <pan>0</pan>
      </midi-instrument>
    </score-part>
    <part-group number="1" type="stop"/>
  </part-list>
  <!--=========================================================-->
  <part id="P1">
    <measure number="1" width="589">
      <print>
        <system-layout>
          <system-margins>
            <left-margin>70</left-margin>
            <right-margin>0</right-margin>
          </system-margins>
          <top-system-distance>211</top-system-distance>
        </system-layout>
        <measure-numbering>system</measure-numbering>
      </print>
      <attributes>
        <divisions>4</divisions>
        <key>
          <fifths>0</fifths>
          <mode>major</mode>
        </key>
        <time>
          <beats>3</beats>
          <beat-type>8</beat-type>
        </time>
        <clef>
          <sign>G</sign>
          <line>2</line>
        </clef>
      </attributes>
      <sound tempo="120"/>
      <note default-x="94">
        <pitch>
          <step>E</step>
          <octave>5</octave>
        </pitch>
        <duration>1</duration>
        <voice>1</voice>
        <type>16th</type>
        <accidental>natural</accidental>
        <stem default-y="-55">down</stem>
        <beam number="1">begin</beam>
        <beam number="2">begin</beam>
      </note>
      <note default-x="176">
        <pitch>
          <step>D</step>
          <alter>1</alter>
          <octave>5</octave>
        </pitch>
        <duration>1</duration>
        <voice>1</voice>
        <type>16th</type>
        <accidental>sharp</accidental>
        <stem default-y="-55">down</stem>
        <beam number="1">continue</beam>
        <beam number="2">continue</beam>
      </note>
      <note default-x="258">
        <pitch>
          <step>E</step>
          <octave>5</octave>
        </pitch>
        <duration>1</duration>
        <voice>1</voice>
        <type>16th</type>
        <stem default-y="-55">down</stem>
        <beam number="1">continue</beam>
        <beam number="2">continue</beam>
      </note>
      <note default-x="341">
        <pitch>
          <step>B</step>
          <octave>4</octave>
        </pitch>
        <duration>1</duration>
        <voice>1</voice>
        <type>16th</type>
        <stem default-y="-55">down</stem>
        <beam number="1">continue</beam>
        <beam number="2">continue</beam>
      </note>
      <note default-x="423">
        <pitch>
          <step>D</step>
          <octave>5</octave>
        </pitch>
        <duration>1</duration>
        <voice>1</voice>
        <type>16th</type>
        <accidental>natural</accidental>
        <stem default-y="-55">down</stem>
        <beam number="1">continue</beam>
        <beam number="2">continue</beam>
      </note>
      <note default-x="505">
        <pitch>
          <step>C</step>
          <octave>5</octave>
        </pitch>
        <duration>1</duration>
        <voice>1</voice>
        <type>16th</type>
        <accidental>natural</accidental>
        <stem default-y="-55">down</stem>
        <beam number="1">end</beam>
        <beam number="2">end</beam>
      </note>
    </measure>
    <!--=======================================================-->
    <measure number="2" width="395">
      <note default-x="13">
        <pitch>
          <step>A</step>
          <octave>4</octave>
        </pitch>
        <duration>2</duration>
        <voice>1</voice>
        <type>eighth</type>
        <stem default-y="10.5">up</stem>
      </note>
      <note default-x="137">
        <rest/>
        <duration>1</duration>
        <voice>1</voice>
        <type>16th</type>
      </note>
      <note default-x="199">
        <pitch>
          <step>C</step>
          <octave>4</octave>
        </pitch>
        <duration>1</duration>
        <voice>1</voice>
        <type>16th</type>
        <stem default-y="5">up</stem>
        <beam number="1">begin</beam>
        <beam number="2">begin</beam>
      </note>
      <note default-x="261">
        <pitch>
          <step>E</step>
          <octave>4</octave>
        </pitch>
        <duration>1</duration>
        <voice>1</voice>
        <type>16th</type>
        <stem default-y="7.5">up</stem>
        <beam number="1">continue</beam>
        <beam number="2">continue</beam>
      </note>
      <note default-x="323">
        <pitch>
          <step>A</step>
          <octave>4</octave>
        </pitch>
        <duration>1</duration>
        <voice>1</voice>
        <type>16th</type>
        <stem default-y="10">up</stem>
        <beam number="1">end</beam>
        <beam number="2">end</beam>
      </note>
      <barline location="right">
        <bar-style>light-heavy</bar-style>
      </barline>
    </measure>
  </part>
  <!--=========================================================-->
  <part id="P2">
    <measure number="1" width="589">
      <print>
        <measure-numbering>system</measure-numbering>
      </print>
      <attributes>
        <divisions>4</divisions>
        <key>
          <fifths>0</fifths>
          <mode>major</mode>
        </key>
        <time>
          <beats>3</beats>
          <beat-type>8</beat-type>
        </time>
        <clef>
          <sign>G</sign>
          <line>2</line>
        </clef>
      </attributes>
      <sound tempo="120"/>
      <note>
        <rest measure="yes"/>
        <duration>6</duration>
        <voice>1</voice>
      </note>
    </measure>
    <!--=======================================================-->
    <measure number="2" width="395">
      <attributes>
        <clef>
          <sign>F</sign>
          <line>4</line>
        </clef>
      </attributes>
      <note default-x="13">
        <pitch>
          <step>A</step>
          <octave>2</octave>
        </pitch>
        <duration>1</duration>
        <voice>1</voice>
        <type>16th</type>
        <stem default-y="-70">down</stem>
        <beam number="1">begin</beam>
        <beam number="2">begin</beam>
      </note>
      <note default-x="75">
        <pitch>
          <step>E</step>
          <octave>3</octave>
        </pitch>
        <duration>1</duration>
        <voice>1</voice>
        <type>16th</type>
        <stem default-y="-67">down</stem>
        <beam number="1">continue</beam>
        <beam number="2">continue</beam>
      </note>
      <note default-x="137">
        <pitch>
          <step>A</step>
          <octave>3</octave>
        </pitch>
        <duration>1</duration>
        <voice>1</voice>
        <type>16th</type>
        <stem default-y="-65">down</stem>
        <beam number="1">end</beam>
        <beam number="2">end</beam>
      </note>
      <note default-x="199">
        <rest/>
        <duration>1</duration>
        <voice>1</voice>
        <type>16th</type>
      </note>
      <note default-x="261">
        <rest/>
        <duration>2</duration>
        <voice>1</voice>
        <type>eighth</type>
      </note>
      <barline location="right">
        <bar-style>light-heavy</bar-style>
      </barline>
    </measure>
  </part>
  <!--=========================================================-->
</score-partwise>
```

**Comment by rettinghaus** [comment]
That the clef is on the *wrong* side of the barline is not a problemn of the MusicXML importer. If you take a look at your XML you'll see, that the clef change is at beginning of the second measure, and not at the end of the previous. 

If you give me the complete XML of the first example I'll have a look at that problem.

**Comment by craigsapp** [comment]
And here is expressing in grand-staff mode in the MusicXML data:

Finale rendering in the editor:

<img width="577" alt="screen shot 2017-04-14 at 12 27 27 pm" src="https://cloud.githubusercontent.com/assets/3487289/25053759/2266f06e-210e-11e7-87d5-a779074304a5.png">

Conversion to SVG with verovio:

<img width="1060" alt="screen shot 2017-04-14 at 12 29 15 pm" src="https://cloud.githubusercontent.com/assets/3487289/25053773/35649036-210e-11e7-9b7b-422d229aada9.png">

Using the command:

```bash
verovio -f musicxml file.xml
```

which matches the behavior first reported.  This means that the musicxml importer into verovio has not yet dealt with the horrors  of grand-staff encoding in MusicXML files....

The MusicXML-to-Humdrum importer has dealt with the horror, and can be used to import into MEI or generate an SVG properly for this example:

```bash
verovio -f musicxml-hum file.xml
```

which renders as:

<img width="761" alt="screen shot 2017-04-14 at 12 32 25 pm" src="https://cloud.githubusercontent.com/assets/3487289/25053864/d0a5cad8-210e-11e7-94d6-f9a8eff37aec.png">

The clef change is still being moved to the wrong side of the barline (due to how it is encoded in the MusicXML).

MusicXML test data:

```xml
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<!DOCTYPE score-partwise PUBLIC "-//Recordare//DTD MusicXML 3.0 Partwise//EN" "http://www.musicxml.org/dtds/partwise.dtd">
<score-partwise version="3.0">
  <movement-title>Fury lease</movement-title>
  <identification>
    <creator type="composer">Louis Beetfield, jr.</creator>
    <rights>©</rights>
    <encoding>
      <software>Finale v25 for Mac</software>
      <encoding-date>2017-04-14</encoding-date>
      <supports attribute="new-system" element="print" type="yes" value="yes"/>
      <supports attribute="new-page" element="print" type="yes" value="yes"/>
      <supports element="accidental" type="yes"/>
      <supports element="beam" type="yes"/>
      <supports element="stem" type="yes"/>
    </encoding>
  </identification>
  <defaults>
    <scaling>
      <millimeters>7.2319</millimeters>
      <tenths>40</tenths>
    </scaling>
    <page-layout>
      <page-height>1545</page-height>
      <page-width>1194</page-width>
      <page-margins type="both">
        <left-margin>140</left-margin>
        <right-margin>70</right-margin>
        <top-margin>70</top-margin>
        <bottom-margin>70</bottom-margin>
      </page-margins>
    </page-layout>
    <system-layout>
      <system-margins>
        <left-margin>0</left-margin>
        <right-margin>0</right-margin>
      </system-margins>
      <system-distance>121</system-distance>
      <top-system-distance>70</top-system-distance>
    </system-layout>
    <staff-layout>
      <staff-distance>80</staff-distance>
    </staff-layout>
    <appearance>
      <line-width type="stem">0.7487</line-width>
      <line-width type="beam">5</line-width>
      <line-width type="staff">0.7487</line-width>
      <line-width type="light barline">0.7487</line-width>
      <line-width type="heavy barline">5</line-width>
      <line-width type="leger">0.7487</line-width>
      <line-width type="ending">0.7487</line-width>
      <line-width type="wedge">0.7487</line-width>
      <line-width type="enclosure">0.7487</line-width>
      <line-width type="tuplet bracket">0.7487</line-width>
      <note-size type="grace">60</note-size>
      <note-size type="cue">60</note-size>
      <distance type="hyphen">120</distance>
      <distance type="beam">8</distance>
    </appearance>
    <music-font font-family="Maestro,engraved" font-size="20.5"/>
    <word-font font-family="Times New Roman" font-size="10.25"/>
  </defaults>
  <credit page="1">
    <credit-type>title</credit-type>
    <credit-words default-x="632" default-y="1475" font-size="24" justify="center" valign="top">Fury lease</credit-words>
  </credit>
  <credit page="1">
    <credit-type>composer</credit-type>
    <credit-words default-x="1122" default-y="1407" font-size="12" justify="right" valign="top">Louis Beetfield, jr.</credit-words>
  </credit>
  <credit page="1">
    <credit-type>rights</credit-type>
    <credit-words default-x="632" default-y="53" font-size="10" justify="center" valign="bottom">©</credit-words>
  </credit>
  <credit page="1">
    <credit-words default-x="140" default-y="1478" font-size="12" valign="top">Score</credit-words>
  </credit>
  <part-list>
    <score-part id="P1">
      <part-name print-object="no">MusicXML Part</part-name>
      <score-instrument id="P1-I1">
        <instrument-name>SmartMusicSoftSynth</instrument-name>
        <virtual-instrument/>
      </score-instrument>
      <midi-device>Bank 1</midi-device>
      <midi-instrument id="P1-I1">
        <midi-channel>1</midi-channel>
        <midi-bank>15489</midi-bank>
        <midi-program>1</midi-program>
        <volume>80</volume>
        <pan>0</pan>
      </midi-instrument>
    </score-part>
  </part-list>
  <!--=========================================================-->
  <part id="P1">
    <measure number="1" width="549">
      <print>
        <system-layout>
          <system-margins>
            <left-margin>70</left-margin>
            <right-margin>0</right-margin>
          </system-margins>
          <top-system-distance>177</top-system-distance>
        </system-layout>
        <measure-numbering>system</measure-numbering>
      </print>
      <attributes>
        <divisions>4</divisions>
        <key>
          <fifths>0</fifths>
          <mode>major</mode>
        </key>
        <time>
          <beats>3</beats>
          <beat-type>8</beat-type>
        </time>
        <staves>2</staves>
        <clef number="1">
          <sign>G</sign>
          <line>2</line>
        </clef>
        <clef number="2">
          <sign>G</sign>
          <line>2</line>
        </clef>
      </attributes>
      <sound tempo="120"/>
      <note default-x="94">
        <pitch>
          <step>E</step>
          <octave>5</octave>
        </pitch>
        <duration>1</duration>
        <voice>1</voice>
        <type>16th</type>
        <accidental>natural</accidental>
        <stem default-y="-55">down</stem>
        <staff>1</staff>
        <beam number="1">begin</beam>
        <beam number="2">begin</beam>
      </note>
      <note default-x="170">
        <pitch>
          <step>D</step>
          <alter>1</alter>
          <octave>5</octave>
        </pitch>
        <duration>1</duration>
        <voice>1</voice>
        <type>16th</type>
        <accidental>sharp</accidental>
        <stem default-y="-55">down</stem>
        <staff>1</staff>
        <beam number="1">continue</beam>
        <beam number="2">continue</beam>
      </note>
      <note default-x="245">
        <pitch>
          <step>E</step>
          <octave>5</octave>
        </pitch>
        <duration>1</duration>
        <voice>1</voice>
        <type>16th</type>
        <stem default-y="-55">down</stem>
        <staff>1</staff>
        <beam number="1">continue</beam>
        <beam number="2">continue</beam>
      </note>
      <note default-x="321">
        <pitch>
          <step>B</step>
          <octave>4</octave>
        </pitch>
        <duration>1</duration>
        <voice>1</voice>
        <type>16th</type>
        <stem default-y="-55">down</stem>
        <staff>1</staff>
        <beam number="1">continue</beam>
        <beam number="2">continue</beam>
      </note>
      <note default-x="397">
        <pitch>
          <step>D</step>
          <octave>5</octave>
        </pitch>
        <duration>1</duration>
        <voice>1</voice>
        <type>16th</type>
        <accidental>natural</accidental>
        <stem default-y="-55">down</stem>
        <staff>1</staff>
        <beam number="1">continue</beam>
        <beam number="2">continue</beam>
      </note>
      <note default-x="472">
        <pitch>
          <step>C</step>
          <octave>5</octave>
        </pitch>
        <duration>1</duration>
        <voice>1</voice>
        <type>16th</type>
        <accidental>natural</accidental>
        <stem default-y="-55">down</stem>
        <staff>1</staff>
        <beam number="1">end</beam>
        <beam number="2">end</beam>
      </note>
      <backup>
        <duration>6</duration>
      </backup>
      <note>
        <rest measure="yes"/>
        <duration>6</duration>
        <voice>2</voice>
        <staff>2</staff>
      </note>
    </measure>
    <!--=======================================================-->
    <measure number="2" width="364">
      <attributes>
        <clef number="2">
          <sign>F</sign>
          <line>4</line>
        </clef>
      </attributes>
      <note default-x="13">
        <pitch>
          <step>A</step>
          <octave>4</octave>
        </pitch>
        <duration>2</duration>
        <voice>1</voice>
        <type>eighth</type>
        <stem default-y="10.5">up</stem>
        <staff>1</staff>
      </note>
      <note default-x="127">
        <rest/>
        <duration>1</duration>
        <voice>1</voice>
        <type>16th</type>
        <staff>1</staff>
      </note>
      <note default-x="184">
        <pitch>
          <step>C</step>
          <octave>4</octave>
        </pitch>
        <duration>1</duration>
        <voice>1</voice>
        <type>16th</type>
        <stem default-y="5">up</stem>
        <staff>1</staff>
        <beam number="1">begin</beam>
        <beam number="2">begin</beam>
      </note>
      <note default-x="240">
        <pitch>
          <step>E</step>
          <octave>4</octave>
        </pitch>
        <duration>1</duration>
        <voice>1</voice>
        <type>16th</type>
        <stem default-y="7.5">up</stem>
        <staff>1</staff>
        <beam number="1">continue</beam>
        <beam number="2">continue</beam>
      </note>
      <note default-x="298">
        <pitch>
          <step>A</step>
          <octave>4</octave>
        </pitch>
        <duration>1</duration>
        <voice>1</voice>
        <type>16th</type>
        <stem default-y="10">up</stem>
        <staff>1</staff>
        <beam number="1">end</beam>
        <beam number="2">end</beam>
      </note>
      <backup>
        <duration>6</duration>
      </backup>
      <note default-x="13">
        <pitch>
          <step>A</step>
          <octave>2</octave>
        </pitch>
        <duration>1</duration>
        <voice>2</voice>
        <type>16th</type>
        <stem default-y="-70">down</stem>
        <staff>2</staff>
        <beam number="1">begin</beam>
        <beam number="2">begin</beam>
      </note>
      <note default-x="70">
        <pitch>
          <step>E</step>
          <octave>3</octave>
        </pitch>
        <duration>1</duration>
        <voice>2</voice>
        <type>16th</type>
        <stem default-y="-67.5">down</stem>
        <staff>2</staff>
        <beam number="1">continue</beam>
        <beam number="2">continue</beam>
      </note>
      <note default-x="127">
        <pitch>
          <step>A</step>
          <octave>3</octave>
        </pitch>
        <duration>1</duration>
        <voice>2</voice>
        <type>16th</type>
        <stem default-y="-65">down</stem>
        <staff>2</staff>
        <beam number="1">end</beam>
        <beam number="2">end</beam>
      </note>
      <note default-x="184">
        <rest/>
        <duration>1</duration>
        <voice>2</voice>
        <type>16th</type>
        <staff>2</staff>
      </note>
      <note default-x="240">
        <rest/>
        <duration>2</duration>
        <voice>2</voice>
        <type>eighth</type>
        <staff>2</staff>
      </note>
      <barline location="right">
        <bar-style>light-heavy</bar-style>
      </barline>
    </measure>
  </part>
  <!--=========================================================-->
</score-partwise>
```

**Comment by rettinghaus** [comment]
Hihi! This is funny, as the clef lands in another layer as the notes. So Verovio renders it correctly in some way, it just *looks* wrong. I'll have a look.

**Comment by craigsapp** [comment]
> That the clef is on the wrong side of the barline is not a problemn of the MusicXML importer. If you take a look at your XML you'll see, that the clef change is at beginning of the second measure, and not at the end of the previous.
>
> If you give me the complete XML of the first example I'll have a look at that problem.

Also note in my encoding example the clef gets placed on the wrong side if the barline (and it is not my fault as that is where Finale is placing it, but it displays the clef in the previous measure as you can see from the screen shot).  

This is not the source of the error however, and instead comes from the need to split the grand-staff part of MusicXML into two separate staves in MEI.

The MusicXML importer is generating two layers for the bottom staff, and placing the clef change in the first layer, and the music in the other layer (due to how the voices in MusicXML are mapped into the staff/layer system of MEI):

```xml
                            <staff xml:id="staff-0000001061635653" n="2">
                                <layer xml:id="layer-0000001616280695" n="1">
                                    <clef xml:id="clef-0000001308989962" shape="F" line="4" />
                                </layer>
                                <layer xml:id="layer-0000002011336583" n="2">
                                    <beam xml:id="beam-0000000993863054">
                                        <note xml:id="note-0000000728542212" dur="16" oct="2" pname="a" stem.dir="down" />
                                        <note xml:id="note-0000001804685537" dur="16" oct="3" pname="e" stem.dir="down" />
                                        <note xml:id="note-0000000290790131" dur="16" oct="3" pname="a" stem.dir="down" />
                                    </beam>
                                    <rest xml:id="rest-0000001784434792" dur="16" />
                                    <rest xml:id="rest-0000001386418789" dur="8" />
                                </layer>
                            </staff>
```

This also gets back to my complaint about how clefs and layers interact in a measure...  What is happening here is that the first layer of the measure is being displayed in bass clef, but the second layer is being displayed (on the same staff) in treble clef.    This is impossible to do in standard music notation syntax, but is easy to do in verovio (such as this example).

In other words, clefs operate on the staff lever, not the layer level.

**Comment by phboer** [comment]
Thanks for looking into this. Attached the full XML file (Fur Elise).

[fur_elise.zip](https://github.com/rism-ch/verovio/files/930519/fur_elise.zip)
